mlrun/runtimes/base.py

```diff
@@ -1394,6 +1392,32 @@ def doc(self):
                             line += f", default={p['default']}"
                         print("    " + line)
 
+    def _encode_requirements(self, requirements_to_encode):
+
+        # if a string, read the file then encode
+        if isinstance(requirements_to_encode, str):
+            with open(requirements_to_encode, "r") as fp:
+                requirements_to_encode = fp.read().splitlines()
+
+        requirements = []
+        for requirement in requirements_to_encode:
+            requirement = requirement.strip()
+            escaped = False
+
+            # -r / --requirement are flags and should not be escaped
+            for req_flag in ["-r", "--requirement"]:
+                if requirement.startswith(req_flag):
+                    requirements.append(
+                        f"{req_flag} '{requirement[len(req_flag):].strip()}'"
```

Comment:
What if the user was already smart and had quotes around the requirement (for example `-r 'reqs.txt'`) ? This will result in double quotes.

Reply:
will `shlex` it